tests/integration/test_hive.py

```diff
@@ -196,3 +211,16 @@ def test_select_partitions(hive_cursor):
     expected_df["j"] = expected_df["j"].astype("int64")
 
     assert_eq(result_df, expected_df, check_index=False)
+
+
+def test_select_multipartitions(hive_cursor):
+    c = Context()
+    c.create_table("df_parts", hive_cursor)
+
+    result_df = c.sql("SELECT * FROM df_parts")
+    expected_df = pd.DataFrame({"i": [1, 2], "j": [1, 2], "k": ["a", "b"]})
+    expected_df["i"] = expected_df["i"].astype("int32")
+    expected_df["j"] = expected_df["j"].astype("int64")
+    expected_df["k"] = expected_df["k"].astype("object")
```

Comment:
We can probably get away without explicitly casting these since they are already the same type, but I don't have a strong opinion either way.

Reply:
These were largely copied from the pre-existing Hive partition case. If anything, I think it makes it clearer at a quick glance exactly what type will be returned.